### DISCUSSION TITLE
`specific_transformers` accepting 2 formats in TableVectorizer

### DISCUSSION THREAD

**Discussion by jeromedockes** [skrub-data/skrub#781] [General]
in TableVectorizer, for `specific_transformers` we can pass a list of (name, transformer, columns) tuples OR a list of (transformer, columns) tuples (but not a list mixing both). this adds complexity in the docstring and parameter validation code, and makes it different from scikit-learn's ColumnTransformer which only accepts the first format.

WDYT about allowing only the first format (ie the same as ColumnTransformer)?

**Comment by GaelVaroquaux** [comment]
> WDYT about allowing only the first format (ie the same as ColumnTransformer)?

No. The ColumnTransformer's API is really clunky to use

**Comment by jeromedockes** [comment]
> No. The ColumnTransformer's API is really clunky to use

only the first format then?

**Reply by GaelVaroquaux** [reply]
using only the first format can make model selection hard because you don't control the name of the corresponding transformer.
